app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+        links = json_data.pop('links', [])
+        if links:
+            json_data['links'] = [link.json for link in self.attr_links]
+        return json_data
+
+    def _get_properties(self):
+        properties = {}
+        for attribute in self.__dict__:
+            if not attribute.startswith('attr_properties_'):
+                continue
+            key = attribute[16:]
+            value = getattr(self, attribute)
+            properties[key] = value
+            if key.endswith('datetime') and isinstance(value, datetime):
+                properties[key] = isoformat(value)
+
+        return properties
+
+    def create(self):
+        '''Create the sample in DB
+
+        Returns:
+            the DB sample (model object)
+        '''
+        attributes = self.get_attributes(remove_relations=False)
+        links = attributes.pop('links', [])
+        self._create_model(attributes)
+        if not self.model_links_instance and links:
+            self._create_model_links(links)
+        return self.model_instance
+
+    @property
+    def model_links(self):
+        '''Returns a django DB model object of the links sample data
+
+        If the data has not yet been created in DB, then it is created.
+
+        Returns:
+            list of link model instances
+        '''
+        if not self.model_links_instance:
+            attributes = self.get_attributes(remove_relations=False)
+            links = attributes.pop('links', [])
+            if not self.model_instance:
+                self._create_model(attributes)
+            self._create_model_links(links)
+        return self.model_links_instance
+
+    def _create_model(self, attributes):
+        attributes.pop('links', None)
+        self.model_instance = Item(**attributes)
+        self.model_instance.full_clean()
+        self.model_instance.save()
+
+    def _create_model_links(self, links):
+        for link in links:
+            link_instance = ItemLink(item=self.model_instance, **link)
+            link_instance.full_clean()
+            link_instance.save()
+            self.model_links_instance.append(link_instance)
+
+
+class AssetSample(SampleData):
+    '''Asset Sample Data
+    '''
+    model_class = Asset
+    sample_name = 'asset'
+    samples_dict = asset_samples
+    key_mapping = {
+        'name': 'id',
+        'eo_gsd': 'eo:gsd',
+        'geoadmin_variant': 'geoadmin:variant',
+        'geoadmin_lang': 'geoadmin:lang',
+        'proj_epsg': 'proj:epsg',
+        'media_type': 'type',
+        'checksum_multihash': 'checksum:multihash'
+    }
+    optional_fields = [
+        'title', 'description', 'eo_gsd', 'geoadmin_variant', 'geoadmin_lang', 'proj_epsg', 'file'
+    ]
+
+    def __init__(self, sample='asset-1', item=None, name=None, **kwargs):
+        '''Create a item sample data
+
+        Args:
+            sample: string
+                Name of the sample based to use, see tests.sample_data.asset_samples
+            required_only: bool
```

Comment:
`required_only` is not an argument of `__init__`, probably docstring has not been updated after changes?